### PR TITLE
Hook up Github api

### LINKED ISSUE #29: Hook up Github API
Connect to the github search api so that we can display query results.

### PR DESCRIPTION
**Pull request by lexahall**
This pull connects the query form on Columns with the github search api so that we can actually display issues/prs!

image

Closes #29

### CODE REVIEW

**Comment by lexahall** [comment]
Going to be adding tests soon

File: src/components/Issue.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    issue.commits.commits[0].commit &&
+    issue.commits.commits[0].commit.status &&
+    issue.commits.commits[0].commit.status.state
+
+  if (buildStatus === 'SUCCESS') {
+    return <CheckIcon color={colors.green[8]} size={16} />
+  }
+
+  if (buildStatus === 'PENDING') {
+    return <CircleIcon color={colors.yellow[8]} size={12} strokeWidth={3} />
```
Comment (colebemis) [inline src/components/Issue.js]: Maybe we should have add a `SmallCircleIcon` so all these icons can have consistent size?

File: src/components/Issue.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    />
+  )
+}
+
+function renderStatusIcon(issue) {
+  const buildStatus =
+    issue.commits &&
+    issue.commits.commits[0].commit &&
+    issue.commits.commits[0].commit.status &&
+    issue.commits.commits[0].commit.status.state
```
Comment (colebemis) [inline src/components/Issue.js]: Maybe we could use [`idx`](https://github.com/facebookincubator/idx) to make this a little cleaner.

File: src/components/Issue.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  if (buildStatus === 'SUCCESS') {
+    return <CheckIcon color={colors.green[8]} size={16} />
+  }
+
+  if (buildStatus === 'PENDING') {
+    return <CircleIcon color={colors.yellow[8]} size={12} strokeWidth={3} />
+  }
+
+  if (buildStatus === 'FAILURE') {
+    return <XIcon color={colors.red[8]} size={16} />
```
Comment (colebemis) [inline src/components/Issue.js]: Would it be better to use a `switch` for these?

File: src/components/Issue.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  }
+}
+
+function Issue({ issue }) {
+  return (
+    <Div
+      display="flex"
```
Comment (colebemis) [inline src/components/Issue.js]: You could use the `Flex` component here.

File: src/components/Issues.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    this.applySearch(this.props.query)
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.query !== this.props.query) {
+      const delay = 500
+
+      clearTimeout(this.setTimeoutId)
+
+      this.setTimeoutId = setTimeout(() => {
```
Comment (colebemis) [inline src/components/Issues.js]: I know you copied this code from me, but I'm wondering if there's a better name for `setTimeoutId`.

File: src/utils/auth.js [hunk 6]
```diff
@@ -19,10 +21,12 @@ export async function logIn(githubCode) {
   })
 
   setAuthToken(data.authenticate.token)
+  setGithubToken(data.authenticate.githubToken)
 }
 
 export function logOut() {
   removeAuthToken()
+  removeGithubToken()
```
Comment (colebemis) [inline src/utils/auth.js]: I think we need to update the tests to take githubToken into account.

File: src/components/Issue.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      padding={spacing[4]}
```
Comment (colebemis) [inline src/components/Issue.js]: The screenshot looked like Issue had a little too much padding. Maybe we should try `spacing[3]`?

File: src/components/Issue.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      borderBottom={`1px solid ${colors.gray[2]}`}
+    >
+      <Div display="flex" flexDirection="column" alignItems="center">
+        <Div marginRight={spacing[4]}>{renderStateIcon(issue)}</Div>
+        <Div marginRight={spacing[4]} marginTop={spacing[1]}>
```
Comment (colebemis) [inline src/components/Issue.js]: Should we move these `marginRight` props to the parent so we don't have to repeat them?

File: src/components/Issue.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+          {renderStatusIcon(issue)}
+        </Div>
+      </Div>
+      <Div display="flex" flexDirection="column" alignItems="flex-start">
+        <IssueTitle to={issue.url}>{issue.title}</IssueTitle>
+        <Small
+          marginTop={spacing[1]}
+          fontSize={fontSizes[0]}
+          lineHeight={lineHeights.tight}
+          color={colors.gray[7]}
```
Comment (colebemis) [inline src/components/Issue.js]: Maybe it's worth breaking this into it's own glamorous component?

**Pull request merged**